src/coord/src/coord.rs

```diff
@@ -2172,7 +2172,12 @@ where
             // original sources on which they depend.
             PeekWhen::Immediately => {
                 if !indexes_complete {
-                    bail!("Unable to automatically determine a timestamp for your query; this can happen if your query depends on non-materialized sources");
+                    bail!(
+                        "Unable to automatically determine a timestamp for your query; \
+                        this can happen if your query depends on non-materialized sources. For
+                        more details, see \
+                        https://materialize.com/docs/sql/create-view/#querying-non-materialized-views"
+                    );
```

Comment:
possibly this `For` should end with a backslash, or should be on its own line in the output?

```suggestion
                    bail!(
                        "Unable to automatically determine a timestamp for your query; \
                        this can happen if your query depends on non-materialized sources.\n\
                        For more details, see \
                        https://materialize.com/docs/sql/create-view/#querying-non-materialized-views"
                    );
```